aiida_quantumespresso/workflows/pdos.py

```diff
@@ -452,10 +456,16 @@ def _generate_dos_inputs(self):
         dos_inputs = AttributeDict(self.exposed_inputs(DosCalculation, 'dos'))
         dos_inputs.parent_folder = self.ctx.nscf_parent_folder
         dos_parameters = self.inputs.dos.parameters.get_dict()
+        fermi_energy_range = self.inputs.fermi_energy_range
 
-        if dos_parameters.pop('align_to_fermi', False):
-            dos_parameters['DOS']['Emin'] = dos_parameters['Emin'] + self.ctx.nscf_fermi
-            dos_parameters['DOS']['Emax'] = dos_parameters['Emax'] + self.ctx.nscf_fermi
+        if fermi_energy_range:
+            dos_parameters['DOS']['Emin'] = fermi_energy_range[0] + self.ctx.nscf_fermi
+            dos_parameters['DOS']['Emax'] = fermi_energy_range[1] + self.ctx.nscf_fermi
+        elif 'Emin' in dos_parameters['DOS'] and 'Emax' in dos_parameters['DOS']:
+            pass
+        else:
+            dos_parameters['DOS']['Emin'] = self.ctx.nscf_emin
+            dos_parameters['DOS']['Emax'] = self.ctx.nscf_emax
```

Comment:
```suggestion
        else:
            dos_parameters['DOS'].setdefault('Emin', self.ctx.nscf_emin)
            dos_parameters['DOS'].setdefault('Emax', self.ctx.nscf_emax)
```
This solution might be a bit more elegant although it is not completely identical since now the keys are decoupled and won;t be updated only if neither are specified. But this may actually be a desirable side effect.

Reply:
Thanks @sphuber. I was not aware of this. I will add this solution.